dotCMS/src/integration-test/java/com/dotcms/enterprise/publishing/remote/DependencyBundlerTest.java

```diff
@@ -0,0 +1,572 @@
+import com.dotcms.util.IntegrationTestInitService;
+import com.dotmarketing.beans.MultiTree;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.containers.model.Container;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.folders.model.Folder;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.structure.model.Relationship;
+import com.dotmarketing.portlets.templates.model.Template;
+import com.dotmarketing.util.Logger;
+import com.dotmarketing.util.UUIDGenerator;
+import com.dotmarketing.util.UtilMethods;
+import com.dotmarketing.util.WebKeys.Relationship.RELATIONSHIP_CARDINALITY;
+import com.liferay.portal.model.User;
+import io.vavr.API;
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.Assert;
+
+public class DependencyBundlerTest extends IntegrationTestBase {
+
+    private static User systemUser;
+    private static String defaultFilterKey = "Intelligent.yml";
+
+    @BeforeClass
+    public static void prepare() throws Exception {
+
+        //Setting web app environment
+        IntegrationTestInitService.getInstance().init();
+        LicenseTestUtil.getLicense();
+
+        systemUser = APILocator.getUserAPI().getSystemUser();
+        createFilterDescriptor(defaultFilterKey,true,true,false,null,null,null,null,true);
+    }
+
+    /**
+     * Creates and saves bundle
+     * @param bundleName
+     * @return a bundle
+     * @throws DotDataException
+     */
+    private static Bundle createBundle (final String bundleName, final boolean forcePush,final String filterKey)
+            throws DotDataException {
+
+        final BundleAPI bundleAPI         = APILocator.getBundleAPI();
+        final Bundle     bundle1           = new Bundle(bundleName, null, null, systemUser.getUserId(),forcePush,filterKey);
+
+        bundleAPI.saveBundle(bundle1);
+        return bundle1;
+    }
+
+    /**
+     * Creates a Filter and adds it to map of filters
+     *
+     * @param key
+     * @param dependencies
+     * @param relationships
+     * @param forcePush
+     * @param excludeClasses
+     * @param excludeDependencyClasses
+     * @param excludeQuery
+     * @param excludeDependencyQuery
+     * @param defaultFilter
+     */
+    private static void createFilterDescriptor(final String key, final boolean dependencies,
+            final boolean relationships, final boolean forcePush,
+            final List<Object> excludeClasses, final List<Object> excludeDependencyClasses,
+            final String excludeQuery, final String excludeDependencyQuery, final boolean defaultFilter){
```

Comment:
Avoid methods with more than 4/5 parameters. It gets too confusing. Create an object instead, probably using the Builder pattern to build it, which give you named methods) and then pass it to this method.